services/background-worker/src/Wire/BackendNotificationPusher.hs

```diff
@@ -78,32 +81,91 @@ pushNotification runningFlag targetDomain (msg, envelope) = do
+            -- dangerous as it could happen that a new type of notification is
+            -- introduced and an old instance of this worker is running, in which case
+            -- the notification will just get dropped. On the other hand not dropping
+            -- this message blocks the whole queue. Perhaps there is a better way to
+            -- deal with this.
+            lift $ reject envelope False
+          Right notif -> do
+            -- FUTUREWORK: Drop support for parsing it as a
+            -- single notification as soon as we can guarantee
+            -- that the message queue does not contain any
+            -- 'BackendNotification's anymore.
+            ceFederator <- asks (.federatorInternal)
+            ceHttp2Manager <- asks http2Manager
+            let ceOriginDomain = notif.ownDomain
+                ceTargetDomain = targetDomain
+                ceOriginRequestId = fromMaybe (RequestId "N/A") notif.requestId
+                cveEnv = FederatorClientEnv {..}
+                cveVersion = Just V0 -- V0 is assumed for non-versioned queue messages
+                fcEnv = FederatorClientVersionedEnv {..}
+            liftIO $ either throwM pure =<< sendNotification fcEnv notif.targetComponent notif.path notif.body
+            lift $ ack envelope
+            metrics <- asks backendNotificationMetrics
+            withLabel metrics.pushedCounter (domainText targetDomain) incCounter
+            withLabel metrics.stuckQueuesGauge (domainText targetDomain) (flip setGauge 0)
+      Right bundle -> do
+        federator <- asks (.federatorInternal)
+        manager <- asks http2Manager
+        let env =
+              FederatorClientEnv
+                { ceOriginDomain = ownDomain . NE.head $ bundle.notifications,
+                  ceTargetDomain = targetDomain,
+                  ceFederator = federator,
+                  ceHttp2Manager = manager,
+                  ceOriginRequestId =
+                    fromMaybe (RequestId "N/A") . (.requestId) . NE.head $ bundle.notifications
+                }
+        -- TODO(md): pull this out into a separate function for redability and testability
+        remoteVersions :: Set Int <-
+          liftIO
+            ( runFederatorClient @'Brig env $
+                fedClientIn @'Brig @"api-version" ()
+            )
+            >>= \case
+              Left e -> do
+                Log.err $
+                  Log.msg (Log.val "Failed to get supported API versions, the notification will be ignored")
+                    . Log.field "domain" (domainText targetDomain)
+                    . Log.field "error" (displayException e)
+                throwM e
+              Right vi -> pure . Set.fromList . fmap versionInt . vinfoSupported $ vi
+        case mostRecentTuple bodyVersions (notifications bundle) remoteVersions of
+          Nothing ->
+            -- TODO(md): do more severe logging warning the site operator
+            Log.err $
```

Comment:
this probably creates a lot of traffic on the log service.  we should probably think about logging this only once every minute or so?  not sure if that's feasible.

Reply:
This is a pretty serious condition, I'm not sure it's worth adding complexity and obfuscating the logs to avoid some potential noise.